gamelauncher.py

```diff
@@ -49,13 +49,36 @@ def parse_args() -> Namespace:  # noqa: D103
 
 
 def _setup_pfx(path: str) -> None:
-    """Create a symlink to the WINE prefix and tracked_files file."""
+    """Create a symlink to the WINE prefix and tracked_files file.
+
+    Also, create a symlink of steamuser to the Unix username
+    """
+    username: str = Path().home().name
+
     if not (Path(path + "/pfx")).expanduser().is_symlink():
         # When creating the symlink, we want it to be in expanded form when passed unexpanded paths
-        # Example: pfx -> /home/.wine
+        # Good: pfx -> /home/foo/.wine
+        # Bad: pfx -> ~/.wine
         Path(path + "/pfx").expanduser().symlink_to(Path(path).expanduser())
     Path(path + "/tracked_files").expanduser().touch()
 
+    # Create a symlink of the current user to the steamuser dir for Steam functionality
```

Comment:
> Personally though, I'm sorta on the fence of this feature because I think we're being too intrusive and it can sorta encourage the practice of re-using WINE prefixes

I agree with you on this one, but according to a previous discussion on Discord around symlinking `<path>/<WINEPREFIX>/pfx` to `<path>/<WINEPREFIX>`, the general consensus was that users will be users and they will mix and match prefixes to wine versions anyway, and keeping compatibility is more important.

> Besides Steam cloud saves, what else do we gain from this?

We do not gain steam cloud saves with this change. Steam cloud saves is part of the Valve opted for hardcoding the user folder to `steamuser`. This is also the reason we cannot patch out this behavior from Proton builds as they are meant to be also usable through Steam AFAIK. We also cannot patch this behavior into other versions of Wine, because they are not controlled by us necessarily, for example Wine from distribution repositories. Hence the need to handle it on the filesystem level.

This discussion stems directly from there, as in this case we need to have common saves files and game configuration in both ULWGL based runners and Wine bases runners, since the prefix is shared.

>  we're deciding for the user's on how their home directory should be structured.

Only their wine home directory, which as long as it works, it shouldn't be something they will ever notice. And as long as we do not overwrite anything, there shouldn't be a problem either.

The one problem I can think of is that a few years back Wine changed its wine home folder structure from Windows XP to Windows Vista, I think this was from Wine 6 to Wine 7, which is ample time to not be considered an issue any more.